scripts/library.sh

```diff
@@ -146,6 +146,24 @@ function wait_until_service_has_external_ip() {
   return 1
 }
 
+# Waits for the endpoint to be routable
+# Parameters: $1 - External ingress IP address `kubectl get svc knative-ingressgateway -n istio-system -o jsonpath="{.status.loadBalancer.ingress[*].ip`
+#             $2 - Hostname `kubectl get route <route_name> -o jsonpath="{.status.domain}"`
+function wait_until_routable() {
+    for i in {1..150}; do  # timeout after 5 minutes
+        local val=$(curl -H "Host: $2" "http://$1" >/dev/null 2>&1)
+        if [[ -z "$val" ]]; then
+            echo -n "."
+            sleep 2
+        else
+            echo "App is now routable"
+            return 0
+        fi                
+    done
+    echo "Timed out waiting for app to be routable"
```

Comment:
nit: keep the logging format, add a linefeed and "ERROR: " to the beginning of the message.

Reply:
done